### PR TITLE
Deduplicate graph traversal for entities

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

The logic for traversing entities is only different in its parameters, so it can be deduplicated

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203363157432084/1203444301722127/f) _(internal)_

## 🚫 Blocked by

- #2207 
- #2208 
- #2223 
- #2226 
- #2229 

## 🔍 What does this change?

- Move traversal logic for entities to function
- Fix a bug to correctly resolve the edge interval. This is not necessarily the link's interval but the intersection of both endpoints. We did never encounter issues so far as without deletion it's not possible to construct this case.

### CODE REVIEW

File: apps/hash-graph/lib/graph/src/subgraph/edges/edge.rs [hunk 1]
```diff
@@ -1,16 +1,48 @@
...
+use serde::{ser::SerializeStruct, Serialize, Serializer};
 use utoipa::{openapi, ToSchema};
 
-#[derive(Debug, Hash, PartialEq, Eq, Serialize)]
-#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[derive(Debug, Hash, PartialEq, Eq)]
 pub struct OutwardEdge<K, E> {
     pub kind: K,
     /// If true, interpret this as a reversed mapping and the endpoint as the source, that is,
     /// instead of Source-Edge-Target, interpret it as Target-Edge-Source
```
Comment (Alfred-Mountfield) [inline apps/hash-graph/lib/graph/src/subgraph/edges/edge.rs:8]: Needs updating

Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/subgraph/edges/edge.rs]: Thanks, removed in [Remove outdated comment](https://github.com/hashintel/hash/pull/2230/commits/7a2c56ddc8fbb51f0baedeb2afaadd757fd3e7b4) and added a doc string in [Add comment to EdgeDirection](https://github.com/hashintel/hash/pull/2230/commits/ddfa958cd8ef57d14382f4f3a009d187a30b3392)

File: apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity.rs [hunk 2]
```diff
@@ -27,12 +27,81 @@ use crate::{
...
+                let edge_interval = edge_entity
+                    .metadata
+                    .temporal_versioning()
+                    .variable_time_interval(time_axis)
+                    .intersect(entity_interval)
+                    .unwrap_or_else(|| {
+                        unreachable!("The edge interval and the entity interval do not overlap")
+                    });
+
+                // reversed `HasLeftEntity` is equivalent to an outgoing link `Entity`
```
Comment (Alfred-Mountfield) [inline apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity.rs:81]: Can we just delete this?

Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity.rs:81]:
```suggestion
```

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**